### ISSUE TITLE
ember-cli-rails does not work with ember 1.9

### ISSUE THREAD

**Issue by jeffreybiles** [rwz/ember-cli-rails#74]
I made an ember-cli-rails app, and it automatically installed ember 1.8.1.  It is a very simple app, and [you can see the repository here](https://github.com/jeffreybiles/ember-cli-rails-demo/).  It runs quite well.

However, when I tried to upgrade to ember 1.9, it stopped working.  [These are the changes I made](https://github.com/jeffreybiles/ember-cli-rails-demo/commit/911d9d0854446fc758bc1899a99bcd9b0be9bae1).  

The following error `ActionController::RoutingError (No route matches [GET] "/javascripts/frontend/vendor.js"):` started to alternate with this big block of text:

```
============================= WARNING! =============================

  Seems like Ember frontend application takes more than 5
  seconds to compile.

  To prevent race conditions consider adjusting build timeout
  configuration in your ember initializer:

    EmberCLI.configure do |config|
      config.build_timeout = 10 # in seconds
    end

  Alternatively, you can set build timeout per application like this:

    EmberCLI.configure do |config|
      config.app :frontend, build_timeout: 10
    end

============================= WARNING! =============================
```

My page never loaded.

Is there a known solution for this?  If not, I'm willing to start looking.  Any pointers on where to dig would be appreciated.

**Comment by rwz** [comment]
What happens when you manually cd into EmberCLI app directory and run `ember build`?

**Comment by jeffreybiles** [comment]
`ember build` has worked continually, but I've gotten some interesting further results

In the small linked app, I was able to get it working.  How?  By unpacking the gemfile and linking directly to it.  But when I linked back to the rubygems gem, it... still worked.  The gem equivalent of kicking the machine.

On the larger app where the problem first arose, it's more complicated but less mysterious.  In app.rb in ember-cli-rails, I set ember_path to just return 'ember', and then my builds started working.  This works, but has the significant disadvantage of requiring that ember-cli be installed either globally or in the directory from which the command is run.

I'm not sure the exact relation to upgrading to 1.9; in the larger app, it was already at 1.9, and it didn't work right out of the box.  In the smaller app, I noticed that it started at 1.8.1 and when I upgraded to 1.9, that's when it stopped working (before mysteriously starting working again).

**Comment by rwz** [comment]
Weird. I assume it's NPM acting weird and having version mismatch all around. Next time you experience something like this, try running the following in an app dir: `rm -rf node_modules bower_components && npm cache clean && npm install && bower install`

**Comment by jeffreybiles** [comment]
Resetting all the npm and bower stuff, sadly, did not help.

On my hacked-up version, on the first try, I see `BuildingBuilding.Building..Building...`, then the usual output of running `ember s` or `ember build`.  However', in the console it gives me `GET http://org.lvh.me:3000/javascripts/enterprise/vendor.js 
org.lvh.me/:62 GET http://org.lvh.me:3000/javascripts/enterprise/enterprise.js 404 (Not Found)`.  When I refresh the page, it works.

On the standard version, it shows no evidence of building, and just outputs `sh: clients/enterprise/node_modules/.bin/ember: No such file or directory` before the long line of errors.  

More clues: running `node_modules/.bin/ember build` from within the clients/enterprise folder works, but running `clients/enterprise/node_modules/.bin/ember build` from the rails app returns an error, saying that I must be within an ember-cli folder.

This led me to replace ember_path with
```
    def ember_path
      @ember_path ||= Pathname.new("node_modules/.bin/ember")
    private
```
It was previously trying to use `app_path.join("node_modules", ".bin", "ember")`, but `exec` had already cd'ed to the app_path, so it was doubling up on the app_path.  This was not a problem on the smaller app because the app_path was '../client', which you can run as many times as you want without it changing the effect.

(I also removed the `path.executable?` check because for some reason, the new path is not recognized as an executable.  However, it does execute, and works (every time except the first, because of the same error as above).

**Comment by jeffreybiles** [comment]
tl;dr: it was doubling up on app_path, but you wouldn't notice it if you had an app_path that followed the pattern of "../ember_folder".

**Comment by jeffreybiles** [comment]
PR associate with this issue: https://github.com/rwz/ember-cli-rails/pull/76

**Comment by rwz** [comment]
Ok, I think I see the problem. Sadly, that means that I'd need to reject your PR, though.

The problem is that when you override that path, the gem assumes you're using the full absolute path. When you use relative path instead, the stuff you described happens.

So, as a quick solution, instead of doing `path: "../client"`, do `path: Rails.root.join("..", "client")`.

We'll make the gem account for that in the next version!

**Comment by jeffreybiles** [comment]
thanks, that sounds like a good solution!

**Comment by Leooo** [comment]
Hello, I'm having the same issue I think: using ember 1.10, and with an ember folder in AssessUp/frontend (while my rails app is in AssessUp/app), I get a "not found" error looking for a file in "javascripts/frontend/vendor" when I firststart the server. After I reload the page several times the error disappears (and ember build in the command line on folder AssessUp/frontend works).

This is my ember initializer:

EmberCLI.configure do |c|
  c.app :frontend, path: Rails.root.join('frontend').to_s#, exclude_ember_deps: ["jquery","handlebars"]
en

and I'm using the last ember-cli-rails version, which includes changes in https://github.com/rwz/ember-cli-rails/commit/d10c8c08a176830387959576c5a9bef0922fe0c2.

Any idea how to solve this? Thanks

**Comment by coladarci** [comment]
@Leooo - while there's still no solution, there's a conversation going on in an open issue here: https://github.com/rwz/ember-cli-rails/issues/120
